signer/src/daemon/httpd.c

```diff
@@ -154,9 +158,24 @@ handle_connection(void *cls, struct MHD_Connection *connection,
             (void*) body, MHD_RESPMEM_PERSISTENT);
         http_status_code = MHD_HTTP_BAD_REQUEST;
     }
-  int ret = MHD_queue_response(connection, http_status_code, response);
-  MHD_destroy_response(response);
-  return ret;
+
+    user = MHD_basic_auth_get_username_password(connection, &pass);
+    fail = ((user == NULL) ||
+            (strcmp(user, ((engine_type *)cls)->config->http_interfaces->interfaces[0].user)) ||
+            (strcmp(pass, ((engine_type *)cls)->config->http_interfaces->interfaces[0].pass)));
```

Comment:
This will crash when user and pass are not set, also we have no way to do it unauthenticated now.